cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/ApiCredentials.java

```diff
@@ -0,0 +1,323 @@
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.REBALANCE_OPERATOR_USERNAME;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.TOPIC_OPERATOR_PASSWORD_KEY;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.TOPIC_OPERATOR_USERNAME;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.TOPIC_OPERATOR_USERNAME_KEY;
+
+/**
+ * Uses information in a Kafka Custom Resource to generate a API credentials configuration file to be used for
+ * authenticating to Cruise Control's REST API.
+ */
+public class ApiCredentials {
+    // Regex to match an entry in Jetty's HashLoginService's file format: username: password, rolename
+    private static final Pattern HASH_LOGIN_SERVICE_PATTERN = Pattern.compile("^[\\w-]+\\s*:\\s*\\w+\\s*,\\s*\\w+\\s*$");
+    private static final List<String> FORBIDDEN_USERNAMES = Arrays.asList(
+            HEALTHCHECK_USERNAME,
+            REBALANCE_OPERATOR_USERNAME,
+            TOPIC_OPERATOR_USERNAME
+    );
+
+    private final String userManagedApiSecretName;
+    private final String userManagedApiSecretKey;
+    private final String namespace;
+    private final String cluster;
+    private final Labels labels;
+    private final CruiseControlSpec specSection;
+    private final OwnerReference ownerReference;
+
+    /**
+     * Constructs the Api Credentials Model for managing API users for Cruise Control API.
+     *
+     * @param namespace         Namespace of the cluster
+     * @param cluster           Name of the cluster to which this component belongs
+     * @param labels            Labels for Cruise Control instance
+     * @param ownerReference    Owner reference for Cruise Control instance
+     * @param specSection       Custom resource section configuring Cruise Control API users
+     */
+    public ApiCredentials(String namespace, String cluster, Labels labels, OwnerReference ownerReference, CruiseControlSpec specSection) {
+        this.namespace = namespace;
+        this.cluster = cluster;
+        this.labels = labels;
+        this.ownerReference = ownerReference;
+        this.specSection = specSection;
+
+        if (validateApiUsersConfig(specSection)) {
+            userManagedApiSecretName = specSection.getApiUsers().getValueFrom().getSecretKeyRef().getName();
+            userManagedApiSecretKey = specSection.getApiUsers().getValueFrom().getSecretKeyRef().getKey();
+        } else {
+            userManagedApiSecretName = null;
+            userManagedApiSecretKey = null;
+        }
+    }
+
+    /**
+     * @return  Returns user-managed API credentials secret name
+     */
+    public String getUserManagedApiSecretName() {
+        return userManagedApiSecretName;
+    }
+
+    /**
+     * @return  Returns user-managed API credentials secret key
+     */
+    /* test */ public String getUserManagedApiSecretKey() {
```

Comment:
It is either `public` or `/* test */`. Not both. The `/* test */` is used to indicate that methods with default access modifiers have the default one because of tests. But if it is public, it makes no sense really.

```suggestion
    public String getUserManagedApiSecretKey() {
```